File: BAC 3 (2023-2024)/Programmation Questions spéciales/Spring/Spring - Exercices/Amazing/gateway/src/main/java/be/vinci/gateway/GatewayController.java
package be.vinci.gateway;

import be.vinci.gateway.exceptions.BadRequestException;
import be.vinci.gateway.exceptions.NotFoundException;
import be.vinci.gateway.models.*;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
public class GatewayController {

    private final GatewayService service;

    public GatewayController(GatewayService service) {
        this.service = service;
    }

    @GetMapping("/products")
    public Iterable<Product> readAllProducts() {
        return service.readAllProducts();
    }

    @GetMapping("/products/{id}")
    public ResponseEntity<Product> readOneProduct(@PathVariable int id) {
        Product product;

        try {
            product = service.readOneProduct(id);
        } catch (NotFoundException e) {
            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
        } catch (BadRequestException e) {
            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
        }

        return new ResponseEntity<>(product, HttpStatus.OK);
    }
}

File: BAC 3 (2023-2024)/Programmation Questions spéciales/Spring/Spring - Démonstrations/Cours 5/replication-exercise/gateway/src/main/kotlin/com/example/gateway/GatewayController.kt
package com.example.gateway

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.server.ResponseStatusException

@RestController
class GatewayController(private val dummyProxy: DummyProxy, private val nodeProxy: NodeProxy) {

    @GetMapping("/")
    fun hello() = runCatching { dummyProxy.hello() }.getOrElse { throw ResponseStatusException(HttpStatus.SERVICE_UNAVAILABLE) }

    @GetMapping("/node")
    fun helloNode() = runCatching { nodeProxy.hello() }.getOrElse { throw ResponseStatusException(HttpStatus.SERVICE_UNAVAILABLE) }

}